server/src/main/java/org/opensearch/index/codec/composite/composite99/Composite99DocValuesWriter.java

```diff
@@ -76,6 +90,76 @@ public Composite99DocValuesWriter(DocValuesConsumer delegate, SegmentWriteState
         for (CompositeMappedFieldType type : compositeMappedFieldTypes) {
             compositeFieldSet.addAll(type.fields());
         }
+
+        boolean success = false;
+        try {
+
+            SegmentInfo segmentInfo = new SegmentInfo(
```

Comment:
nit : why not move 97  - 118 to separate method as `getSegmentWriteState(segmentInfo, segmentWriteState)` - keep this one simple